### PR TITLE
Add documentation / manual pages about hardware interfaces

### PR DESCRIPTION
**Pull request by pippin88**
Add documentation / manual pages about hardware interfaces

Currently the documents do not contain sufficient information on actual hardware interfaces between the LinuxCNC computer and the CNC machine. So I made this summary page.

I hope that I have done this correctly to enable easy merging.

### CODE REVIEW

File: docs/src/getting-started/hardware-interface.adoc [hunk 1]
```diff
@@ -1 +1,165 @@
...
+Realtime (time critical) tasks such as step generation are done in software on the LinuxCNC host - this means the parallel port interface is much more sensitive to the LinuxCNC computer's latency.
+
+__Advantages:__
+- Low cost
+- Simple configuration
+
+__Disadvantages:__
+- Sensitive to the LinuxCNC computer's latency
+- Limited inputs/outputs 
+- Some PCI/PCIe parallel port cards do not work well or do not properly support the required EPP mode
```
Comment (andypugh) [inline docs/src/getting-started/hardware-interface.adoc:44]: EPP is not a requirement for p-port step generation etc. It is only required to use the Mesa and Pico cards.

**Comment by andypugh** [comment]
Looks good to me so far, though see my comment about EPP parport. 

I didn't see anything about Modbus in there (but it was a lot of commits to look through)

**Comment by petterreinholdtsen** [comment]
[andypugh]
> Looks good to me so far, though see my comment about EPP parport. 
>
> I didn't see anything about Modbus in there (but it was a lot of
> commits to look through)

I hope the commits will be squashed before merging.  Perhaps rebase and
squash in your branch to get a simpler commit message?
-- 
Happy hacking
Petter Reinholdtsen

**Comment by andypugh** [comment]
I think I can squash when merging, (squash and merge is one of the options in the big green button)

**Comment by pippin88** [comment]
Changed EPP mode comment.

Apologies about the many commits. I tried to find a way to merge commits but failed. I cannot find how to squash / rebase / merge).

I am working on github website, but the options that should be there to squash / merge don't seem to be.

Maybe because I forked LinuxCNC and am working in a fork of the master branch? Instead of creating a new branch?

**Comment by andypugh** [comment]
Do you think that a mention of Modbus should be in this section? 
(There are several options:
http://linuxcnc.org/docs/stable/html/drivers/mb2hal.html
http://linuxcnc.org/docs/stable/html/drivers/mesa_modbus.html
http://linuxcnc.org/docs/stable/html/ladder/classic-ladder.html#_modbus_settings
And a bunch of dedicated VFD drivers that generally work through a USB to serial dongle)

**Comment by pippin88** [comment]
I mention modbus under usb-modbus.

The aim of this page is to explain hardware interfaces from linuxcnc computer to outside world. I.e. how do we get signals into and out of the computer

So Mesa card modbus functions really are an extension of Mesa ethernet / parallel / spi interfaces. 

The new linuxcnc user is often unclear on what they need between computer and stepper drivers or computer and breakout board. The aim of this page is explain options for that.

I don't think enough computers have native modbus interface to have a big section on it.

**Comment by hansu** [comment]
>I don't think enough computers have native modbus interface to have a big section on it.

For running Modbus with Mb2Hal you only need a serial port or a USB-to-serial converter...

**Comment by petterreinholdtsen** [comment]
[pippin88]
> Apologies about the many commits. I tried to find a way to merge
> commits but failed. I cannot find how to squash / rebase / merge).

If you approach me (pere) on IRC (#linuxcnc or #linuxcnc-devel via
irc.libera.chat, I can perhaps provide some guidance.  In short, on the
command line it is a question of using 'git rebase -i' to modify the
commits in your branch.

> I am working on github website, but the options that should be there
> to squash / merge don't seem to be.

I have no idea how to do this via the github web site, and no idea if it
is possible.

> Maybe because I forked LinuxCNC and am working in a fork of the master
> branch? Instead of creating a new branch?

That is not the reason.  A separate branch will save you some troubles,
but it is not the cause of any rebase/squash problems.  You can always
rename your current master to some new name and create a new master
branch identical to the upstream one.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by hansu** [comment]
Andy, did you use the GitHub squash and merge option in the big green button?

**Comment by andypugh** [comment]
I did. But I didn't check if it had worked.

**Comment by hansu** [comment]
I only wondered because it appears as a single commit from pippin88 instead of a merge commit. Interesting...

**Pull request merged**